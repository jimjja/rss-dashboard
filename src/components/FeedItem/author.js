import React from 'react';
import PropTypes from 'prop-types';

const authorStyle = {
  margin: '0 .5em 0 0',
  display: 'inline-block',
  fontWeight: 'normal',
};

const Author = ({ author }) => <h4 style={authorStyle}>{author}</h4>;

Author.propTypes = {
  author: PropTypes.string.isRequired,
};
export default Author;
