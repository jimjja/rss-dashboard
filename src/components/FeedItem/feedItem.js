import React from "react";
import PropTypes from "prop-types";

//Components
import Author from "./author";
import PublishDate from "./publishDate";

const titleStyle = {
  fontSize: 18,
  fontWeight: "bold",
  textDecoration: "none",
  color: "#000000a6"
};

const wrapperStyle = {
  border: "1px solid #ebedf0",
  borderRight: 0,
  borderLeft: 0,
  borderTop: 0,
  padding: "15px",
  color: "#000000a6"
};
const htmlContentStyle = { margin: "1em 0", fontSize: 12 };

const authorStyle = {
  margin: 0
};

const FeedItem = props => {
  const { link, title, pubDate, author, content } = props;
  return (
    <div style={wrapperStyle}>
      <a style={titleStyle} href={link}>
        {title}
      </a>
      <div style={authorStyle}>
        {author && <Author author={author} />}
        {pubDate && <PublishDate pubDate={pubDate} />}
      </div>
      <div
        style={htmlContentStyle}
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </div>
  );
};

// FeedItem.propTypes = {
//   url: PropTypes.string,
//   content: PropTypes.string,
//   title: PropTypes.string.isRequired,
//   pudDate: PropTypes.any,
//   author: PropTypes.shape({
//     name: PropTypes.string,
//   }),
//   contentSnippet: PropTypes.string,
// }

// FeedItem.defaultProps = {
//   contentSnippet: '',
//   url: '',
//   content: '',
//   author: null,
//   pubDate: null,
// }

export default FeedItem;
