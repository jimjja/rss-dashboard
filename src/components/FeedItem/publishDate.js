import React from 'react';
import PropTypes from 'prop-types';

const datePublishedStyle = {
  margin: 0,
  display: 'inline-block',
  fontSize: 12,
};

const PublishDate = ({ pubDate }) => (
  <p style={datePublishedStyle}>
    {`${new Date(pubDate).toLocaleDateString()}`}
  </p>
);

PublishDate.propTypes = {
  pubDate: PropTypes.string.isRequired,
};
export default PublishDate;
