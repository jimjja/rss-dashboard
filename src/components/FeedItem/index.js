import FeedItem from "./feedItem";

export default FeedItem;
