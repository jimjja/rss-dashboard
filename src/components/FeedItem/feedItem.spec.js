import React from 'react';
import renderer from 'react-test-renderer';
import FeedItem from './feedItem';

describe('Feed Item component', () => {
  it('should match snapshot', () => {
    const now = new Date();
    const props = {
      pubDate: now.toString(),
      link: 'http://google.co.uk',
      content: '<h1>I am HTML in string </h1>',
      author: 'Example author',
      title: 'Example title',
    };
    const tree = renderer.create(<FeedItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render author when no value', () => {
    const now = new Date();
    const props = {
      pubDate: now.toString(),
      link: 'http://google.co.uk',
      content: '<h1>I am HTML in string </h1>',
      title: 'Example title',
    };
    const tree = renderer.create(<FeedItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render publish date when no value', () => {
    const props = {
      link: 'http://google.co.uk',
      content: '<h1>I am HTML in string </h1>',
      title: 'Example title',
    };
    const tree = renderer.create(<FeedItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
