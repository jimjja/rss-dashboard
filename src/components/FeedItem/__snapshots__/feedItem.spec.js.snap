// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed Item component should match snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ebedf0",
      "borderLeft": 0,
      "borderRight": 0,
      "borderTop": 0,
      "color": "#000000a6",
      "padding": "15px",
    }
  }
>
  <a
    href="http://google.co.uk"
    style={
      Object {
        "color": "#000000a6",
        "fontSize": 18,
        "fontWeight": "bold",
        "textDecoration": "none",
      }
    }
  >
    Example title
  </a>
  <div
    style={
      Object {
        "margin": 0,
      }
    }
  >
    <h4
      style={
        Object {
          "display": "inline-block",
          "fontWeight": "normal",
          "margin": "0 .5em 0 0",
        }
      }
    >
      Example author
    </h4>
    <p
      style={
        Object {
          "display": "inline-block",
          "fontSize": 12,
          "margin": 0,
        }
      }
    >
      2019-2-24
    </p>
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>I am HTML in string </h1>",
      }
    }
    style={
      Object {
        "fontSize": 12,
        "margin": "1em 0",
      }
    }
  />
</div>
`;

exports[`Feed Item component should not render author when no value 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ebedf0",
      "borderLeft": 0,
      "borderRight": 0,
      "borderTop": 0,
      "color": "#000000a6",
      "padding": "15px",
    }
  }
>
  <a
    href="http://google.co.uk"
    style={
      Object {
        "color": "#000000a6",
        "fontSize": 18,
        "fontWeight": "bold",
        "textDecoration": "none",
      }
    }
  >
    Example title
  </a>
  <div
    style={
      Object {
        "margin": 0,
      }
    }
  >
    <p
      style={
        Object {
          "display": "inline-block",
          "fontSize": 12,
          "margin": 0,
        }
      }
    >
      2019-2-24
    </p>
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>I am HTML in string </h1>",
      }
    }
    style={
      Object {
        "fontSize": 12,
        "margin": "1em 0",
      }
    }
  />
</div>
`;

exports[`Feed Item component should not render publish date when no value 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ebedf0",
      "borderLeft": 0,
      "borderRight": 0,
      "borderTop": 0,
      "color": "#000000a6",
      "padding": "15px",
    }
  }
>
  <a
    href="http://google.co.uk"
    style={
      Object {
        "color": "#000000a6",
        "fontSize": 18,
        "fontWeight": "bold",
        "textDecoration": "none",
      }
    }
  >
    Example title
  </a>
  <div
    style={
      Object {
        "margin": 0,
      }
    }
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>I am HTML in string </h1>",
      }
    }
    style={
      Object {
        "fontSize": 12,
        "margin": "1em 0",
      }
    }
  />
</div>
`;
